crud/stats/module.lua

```diff
@@ -0,0 +1,262 @@
+---- CRUD statistics module
+-- @module crud.stats.module
+--
+
+local clock = require('clock')
+local checks = require('checks')
+local errors = require('errors')
+local vshard = require('vshard')
+
+local dev_checks = require('crud.common.dev_checks')
+local utils = require('crud.common.utils')
+local op_module = require('crud.stats.operation')
+local registry = require('crud.stats.local_registry')
+local stash = require('crud.stats.stash')
+
+local StatsError = errors.new_class('StatsError', {capture_stack = false})
+
+local stats = {}
+local internal = stash.get('internal')
+
+--- Check if statistics module was enabled.
+--
+-- @function is_enabled
+--
+-- @treturn boolean Returns `true` or `false`.
+--
+function stats.is_enabled()
+    return internal.is_enabled == true
+end
+
+--- Initializes statistics registry, enables callbacks and wrappers.
+--
+--  If already enabled, do nothing.
+--
+-- @function enable
+--
+-- @treturn boolean Returns `true`.
+--
+function stats.enable()
+    if stats.is_enabled() then
+        return true
+    end
+
+    StatsError:assert(
+        rawget(_G, 'crud') ~= nil,
+        'Can be enabled only on crud router'
+    )
+
+    internal.is_enabled = true
+    registry.init()
+
+    return true
+end
+
+--- Resets statistics registry.
+--
+--  After reset collectors are the same as right
+--  after initial `stats.enable()`.
+--
+-- @function reset
+--
+-- @treturn boolean Returns true.
+--
+function stats.reset()
+    if not stats.is_enabled() then
+        return true
+    end
+
+    registry.destroy()
+    registry.init()
+
+    return true
+end
```

Comment:
If it wasn't enabled, `reset` should not enable it.

Reply:
I had no an opinion here and asked to understand your vision. `crud.reset_stats()` in its current implementation has a use case: reset counters if statistics is enabled. My variant has no value, because `enable` + `disable` can be called explicitly. So I tend to agree with you.